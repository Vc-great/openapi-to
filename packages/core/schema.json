{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "@openapi-to/define-config",
  "title": "JSON schema for @openapi-to/core define-config",
  "type": "object",
  "required": [],
  "properties": {
    "root": {
      "type": "string",
      "description": "Project root directory. Can be an absolute path, or a path relative from the location of the config file itself.",
      "default": "process.cwd()"
    },
    "input": {
      "type": "object",
      "description": "",
      "properties": {
        "path": {
          "type": "string",
          "description": "Path to be used as the input. Can be an absolute path, or a path relative from the defined root option."
        }
      }
    },
    "output": {
      "type": "object",
      "description": "",
      "properties": {
        "path": {
          "type": "string",
          "description": "Path to be used to export all generated files. Can be an absolute path, or a path relative based of the defined root option."
        },
        "clean": {
          "type": "boolean",
          "description": "Remove previous generated files and folders."
        },
        "write": {
          "type": "boolean",
          "description": "Enabled or disable the writing to the filesystem. This is being used for the playground.",
          "default": "true"
        }
      }
    },
    "hooks": {
      "type": "object",
      "description": "Hooks that can be called when a specific action is done in openapi-to. ",
      "properties": {
        "done": {
          "oneOf": [
            {
              "type": "string",
              "description": "Hook that will be called at the end of all executions. Useful for running Prettier or ESLint to use your own linting structure."
            },
            {
              "type": "array",
              "items": {
                "type": "string"
              },
              "description": "Hook that will be called at the end of all executions. Useful for running Prettier or ESLint to use your own linting structure."
            }
          ]
        }
      }
    },
    "plugins": {
      "oneOf": [
        {
          "type": "array",
          "items": {
            "type": "array",
            "prefixItems": [
              {
                "type": "string"
              }
            ],
            "items": {
              "type": "object"
            }
          }
        },
        {
          "type": "object"
        }
      ],
      "description": "Plugins"
    }
  }
}
