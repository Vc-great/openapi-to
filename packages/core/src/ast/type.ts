import type { JSDocTagStructure } from "ts-morph";
import type {
  AmbientableNodeStructure,
  ExportableNodeStructure,
  JSDocableNodeStructure,
  Structure,
} from "ts-morph";
import type {
  BindingNamedNodeStructure,
  ExclamationTokenableNodeStructure,
  KindedStructure,
  OptionalKind,
  StructureKind,
  TypedNodeStructure,
  VariableDeclarationKind,
  WriterFunction,
} from "ts-morph";

export interface VariableStatementStructureOfObject
  extends Structure,
    VariableStatementSpecificStructure,
    JSDocableNodeStructure,
    AmbientableNodeStructure,
    ExportableNodeStructure {}

interface VariableStatementSpecificStructure
  extends KindedStructure<StructureKind.VariableStatement> {
  declarationKind?: VariableDeclarationKind;
  declarations: OptionalKind<VariableDeclarationStructure>[];
}

export interface VariableDeclarationStructure
  extends Structure,
    VariableDeclarationSpecificStructure,
    BindingNamedNodeStructure,
    InitializerExpressionableNodeStructure,
    TypedNodeStructure,
    ExclamationTokenableNodeStructure {}

interface VariableDeclarationSpecificStructure
  extends KindedStructure<StructureKind.VariableDeclaration> {}

export interface InitializerExpressionableNodeStructure {
  initializer?: string | WriterFunction | ObjectStructure[];
}

export interface ObjectStructure {
  key: string;
  value: string;
  docs: JSDocStructure[];
}

export interface JSDocStructure {
  /**
   * The description of the JS doc.
   * @remarks To force this to be multi-line, add a newline to the front of the string.
   */
  description?: string;
  /** JS doc tags (ex. `&#64;param value - Some description.`). */
  tags?: OptionalKind<JSDocTagStructure>[];
}

export type JSDoc = JSDocStructure[];
