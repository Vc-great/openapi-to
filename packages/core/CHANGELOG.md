# @openapi-to/core

## 3.0.1

### Patch Changes

- fix bug

## 3.0.0

### Major Changes

- fix bug
- fix: bug
- update
- refactor

## 3.0.0-alpha.5

### Major Changes

- fix: bug

## 3.0.0-alpha.4

### Major Changes

- fix bug

## 3.0.0-alpha.3

### Major Changes

- refactor

## 3.0.0-alpha.2

### Major Changes

- fix

## 2.4.0-alpha.1

### Patch Changes

- fix bug

## 2.4.0-alpha.0

### Minor Changes

- update all

## 2.3.0

### Minor Changes

- generate swr

## 2.2.0

### Minor Changes

- fix bug

## 2.1.0

### Minor Changes

- optimize request type zod

## 2.0.0

### Major Changes

- [`e519466`](https://github.com/Vc-great/openapi-to/commit/e5194667c7416e817a498d592c357a7ae9c05f22) Thanks [@Vc-great](https://github.com/Vc-great)! - v2

- [`a39a900`](https://github.com/Vc-great/openapi-to/commit/a39a9002dda434d8a65768f55c69875ed8ad1eea) - fix bug

- 2.0

- [`367be25`](https://github.com/Vc-great/openapi-to/commit/367be252aa434487c09c4566e77792839867b509) - fix bug

- [`7fd9e58`](https://github.com/Vc-great/openapi-to/commit/7fd9e58417ef5563dedf945fbb030b70b8b09bd6) - rc

## 2.0.0-rc.5

### Major Changes

- rc

## 2.0.0-alpha.4

### Major Changes

- fix bug

## 2.0.0-alpha.3

### Major Changes

- fix bug

## 2.0.0-alpha.2

### Major Changes

- v2
