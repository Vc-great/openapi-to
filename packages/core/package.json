{
  "name": "@openapi-to/core",
  "version": "2.4.0-alpha.1",
  "description": "Generator core",
  "keywords": [
    "typescript",
    "plugins",
    "codegen"
  ],
  "repository": {
    "type": "git",
    "url": "",
    "directory": "packages/core"
  },
  "license": "MIT",
  "author": "",
  "sideEffects": false,
  "type": "module",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs",
      "default": "./dist/index.cjs"
    },
    "./utils": {
      "import": "./dist/utils.js",
      "require": "./dist/utils.cjs",
      "default": "./dist/utils.cjs"
    },
    "./package.json": "./package.json",
    "./*": "./*"
  },
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "src",
    "dist",
    "*.d.ts",
    "*.d.cts",
    "schema.json",
    "!/**/**.test.**",
    "!/**/__tests__/**"
  ],
  "scripts": {
    "build": "tsup",
    "clean": "npx rimraf ./dist",
    "lint": "bun biome lint .",
    "lint:fix": "bun biome lint --write --unsafe .",
    "release": "pnpm publish --no-git-check",
    "release:canary": "bash ../../.github/canary.sh && node ../../scripts/build.js canary && pnpm publish --no-git-check -tag canary",
    "release:alpha": "bash ../../.github/canary.sh && node ../../scripts/build.js alpha && pnpm publish --no-git-check -tag alpha",
    "start": "tsup --watch",
    "test": "vitest --passWithNoTests",
    "typecheck": "tsc -p ./tsconfig.json --noEmit --emitDeclarationOnly false"
  },
  "dependencies": {
    "@types/json-schema": "^7.0.15",
    "axios": "^1.6.7",
    "code-block-writer": "^12.0.0",
    "consola": "^3.2.3",
    "directory-tree": "^3.5.1",
    "do-swagger2openapi": "^7.0.18",
    "find-up": "^7.0.0",
    "fs-extra": "^11.2.0",
    "is-chinese": "^2.0.0",
    "is-url": "^1.2.4",
    "js-runtime": "^0.0.7",
    "json-schema": "^0.4.0",
    "lodash": "^4.17.21",
    "natural-orderby": "^3.0.2",
    "oas": "^24.9.0",
    "picocolors": "^1.0.0",
    "pinyin-pro": "^3.25.0",
    "seedrandom": "^3.0.5",
    "semver": "^7.5.4",
    "tinyrainbow": "^1.2.0",
    "ts-morph": "^23.0.0"
  },
  "devDependencies": {
    "@openapi-to/config-ts": "workspace:*",
    "@openapi-to/config-tsup": "workspace:*",
    "@types/fs-extra": "^11.0.4",
    "@types/is-url": "^1.2.32",
    "@types/lodash": "^4.17.9",
    "@types/seedrandom": "^3.0.8",
    "@types/semver": "^7.5.6",
    "ora": "^7.0.1"
  },
  "engines": {
    "node": ">=18",
    "pnpm": ">=8.3.0"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org/"
  }
}
