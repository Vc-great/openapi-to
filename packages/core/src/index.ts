export * from "./defineConfig";
export * from "./errors.ts";
export { Generator } from "./Generator.ts";
export { PackageManager } from "./PackageManager.ts";
// dprint-ignore
export { SchemaGenerator } from "./SchemaGenerator.ts";
export * from "./types.ts";
/*
export default build;*/
export { AST } from "./ast/ast.ts";
export { build, requestRemoteData } from "./build.ts";
export { OpenAPI } from "./openapi/OpenAPI.ts";
export { createPlugin } from "./plugins.ts";

/*



export interface _Register {}
export type Plugins = _Register;
export type OptionsPlugins = { [K in keyof Plugins]: Plugins[K]["options"] };
* */
