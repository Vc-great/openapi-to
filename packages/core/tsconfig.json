{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "types": ["vitest/globals", "bun-types"]
  },
  "include": ["src/**/*.ts", "./package.json", "vitest.config.ts", "tsup.config.ts", "../config-ts/reset.d.ts"]
}
