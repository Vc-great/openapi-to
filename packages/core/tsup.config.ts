import { optionsCJS, optionsESM } from "@openapi-to/config-tsup";

import { defineConfig } from "tsup";

export default defineConfig([
  {
    ...optionsCJS,
    noExternal: [/find-up/],
  },
  optionsESM,
  {
    ...optionsCJS,
    entry: {
      utils: "src/utils/index.ts",
    },
    name: "utils",
    noExternal: [/find-up/],
  },
  {
    ...optionsESM,
    entry: {
      utils: "src/utils/index.ts",
    },
    name: "utils",
    noExternal: [/find-up/],
  },
]);
