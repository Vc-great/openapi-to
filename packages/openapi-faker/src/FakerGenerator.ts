import path from "node:path";

import _ from "lodash";
import { VariableDeclarationKind } from "ts-morph";

import { Component } from "./Component.ts";
import { Schema } from "./Schema.ts";

import type { PluginContext } from "@openapi-to/core";
import type { Operation } from "oas/operation";
import type OasTypes from "oas/types";
import type { ClassDeclarationStructure } from "ts-morph";
import type {
  ImportDeclarationStructure,
  MethodDeclarationStructure,
  OptionalKind,
} from "ts-morph";
import type { Config } from "./types.ts";

type ImportStatementsOmitKind = Omit<ImportDeclarationStructure, "kind">;

type ResponseObject = {
  code: string;
  jsonSchema?: {
    description?: string;
    label: string;
    schema: OasTypes.SchemaObject;
    type: string | string[];
  };
};

export class FakerGenerator {
  private operation: Operation | undefined;
  private oas: Config["oas"];
  private readonly paramsZodSchema: string;
  private readonly openapi: Config["openapi"];
  private readonly ast: Config["ast"];
  private readonly pluginConfig: Config["pluginConfig"];
  private readonly openapiToSingleConfig: Config["openapiToSingleConfig"];
  private importCache: Set<string> = new Set<string>();
  private schema: Schema;
  private component: Component;
  constructor(config: Config) {
    this.oas = config.oas;
    this.ast = config.ast;
    this.pluginConfig = config.pluginConfig;
    this.openapiToSingleConfig = config.openapiToSingleConfig;
    this.openapi = config.openapi;
    this.paramsZodSchema = "paramsZodSchema";

    this.component = new Component(config);
    this.schema = new Schema(config);
  }

  get currentTagName() {
    return _.camelCase(
      this.openapi &&
        this.openapi.currentTagMetadata &&
        this.openapi.currentTagMetadata.name,
    );
  }

  get className() {
    return _.upperFirst(this.currentTagName) + "Faker";
  }

  get lowerFirstClassName() {
    return _.lowerFirst(this.className);
  }

  get namespaceTypeName(): string {
    return _.upperFirst(this.currentTagName);
  }

  build(context: PluginContext): void {
    _.forEach(_.values(this.openapi.pathGroupByTag), (pathGroup) => {
      const methodsStatements = _.map(pathGroup, ({ path, method, tag }) => {
        this.operation = this.openapi.setCurrentOperation(path, method, tag);
        return this.generatorMethod();
      });
      const filePath = path.resolve(
        this.openapiToSingleConfig.output,
        this.lowerFirstClassName + ".ts",
      );
      const refKey = [...this.openapi.refCache.keys()];
      return this.ast.createSourceFile(filePath, {
        statements: [
          ...this.generateImport(refKey),
          this.generatorClass(methodsStatements),
          ...this.generatorExport(),
        ],
      });
      this.openapi.resetRefCache();
    });

    this.component.generateComponentType();
    this.component.generateModelIndex();
  }

  /**
   * @example
   * ```
   * const apiName = new ApiName
   * export { apiName }
   * ```
   */
  generatorExport() {
    return [
      this.ast.generateVariableStatements({
        declarationKind: VariableDeclarationKind.Const,
        declarations: [
          {
            name: this.lowerFirstClassName,
            initializer: "new" + " " + this.className,
          },
        ],
        isExported: true,
      }),
      /*      this.ast.generateExportDeclarationStatements({
        kind: StructureKind.ExportDeclaration,
        namedExports: "apiName",
      }),*/
    ];
  }

  /**
   *
   * @example
   * ```
   * import { faker } from '@faker-js/faker';
   * import type {Pet} from "./Pet";
   * import {list} from './fakerModels'
   * ```
   */
  generateImport(
    importModel: Array<string>,
  ): Array<ImportDeclarationStructure> {
    const model = _.chain(importModel)
      .map(($ref: string) => _.camelCase(this.openapi.getRefAlias($ref)))
      .value();
    const fakerModels: ImportStatementsOmitKind = {
      namedImports: [...model],
      moduleSpecifier: "./fakerModels",
    };

    const faker: ImportStatementsOmitKind = {
      namedImports: ["faker"],
      moduleSpecifier: "@faker-js/faker",
    };

    const typeModel: ImportStatementsOmitKind = {
      isTypeOnly: true,
      namedImports: [this.namespaceTypeName],
      moduleSpecifier: `./${this.namespaceTypeName}`,
    };

    const statements = _.chain([] as Array<ImportStatementsOmitKind>)
      .concat(faker)
      .concat(fakerModels)
      .concat(typeModel)
      .value();

    return this.ast.generateImportStatements(statements);
  }

  generatorClass(
    methodsStatements: OptionalKind<MethodDeclarationStructure>[],
  ): ClassDeclarationStructure {
    const statements = {
      name: this.className,
      docs: [
        {
          description: "\n",
          tags: [
            {
              tagName: "tag",
              text:
                this.openapi &&
                this.openapi.currentTagMetadata &&
                this.openapi.currentTagMetadata.name,
            },
            {
              tagName: "description",
              text:
                this.openapi &&
                this.openapi.currentTagMetadata &&
                this.openapi.currentTagMetadata.description,
            },
          ],
        },
      ],
      methods: methodsStatements,
    };
    return this.ast.generateClassStatements(statements);
  }

  /**
   * @example
   * ```
   * ApiType.FindByIdResponse
   * ```
   */
  generatorReturnType(): string {
    return `NonNullable<${this.namespaceTypeName}.${this.openapi.upperFirstResponseName}>`;
  }

  /**
   * @example
   * ```
   *  @summary 查询事项目录详情
   *  @description 详细描述
   * ```
   */
  generatorMethodDocs() {
    return [
      {
        description: "\n",
        tags: [
          {
            tagName: "summary",
            text: this.operation?.getSummary(),
          },
          {
            tagName: "description",
            text: this.operation?.getDescription(),
          },
        ],
      },
    ];
  }

  /**
   * @example
   * ```
   * request({
   *  method:"get",
   *  headers:{
   *    'Content-type':'application/x-www-form-urlencoded'
   *  },
   *  url:"/xxx/xx",
   *  params:query,
   *  data:body,
   * paramsSerializer(params: BusinessType.DetailByIdQueryRequest) {
   *                 return qs.stringify(params)
   *             }
   * })
   * ```
   */
  generatorMethodBody(): string {
    const codes = this.openapi.response?.getResponseStatusCodes || [];

    const successCode = (codes || []).filter((code) =>
      /^(2[0-9][0-9]|300)$/.test(code),
    );

    const responseObject = _.chain([...successCode])
      .map((code) => {
        return {
          code,
          jsonSchema:
            _.head(this.openapi.response?.getResponseAsJSONSchema(code)) ||
            null,
        };
      })
      .filter((x) => !_.isNull(x.jsonSchema))
      .value() as Array<ResponseObject>;

    return _.isEmpty(responseObject)
      ? "return {}"
      : this.generateFakerSingleSchema(
          _.head(responseObject) as ResponseObject,
        );
  }

  generateFakerSingleSchema({ code, jsonSchema }: ResponseObject): string {
    const schema = jsonSchema?.schema;
    const description = jsonSchema?.description;
    const isError = /^([3-5][0-9][0-9])$/.test(code);
    const name = this.openapi.requestName + "Response" + (isError ? code : "");

    if (!schema) {
      return "return {}";
    }

    if (this.openapi.isReference(schema)) {
      return `return ${this.openapi.getRefAlias(schema.$ref)}()`;
    }

    if (schema.type === "array") {
      //todo
      return `return ${this.schema.formatterSchemaType(schema)}`;
    }
    //todo
    if ("additionalProperties" in schema) {
      return `return {}`;
    }

    return `return ${this.schema.getStatementsFromSchema(schema)}`;
  }

  generatorMethod(): MethodDeclarationStructure {
    const statement = {
      name: this.openapi.requestName,
      decorators: [],
      parameters: [],
      returnType: this.generatorReturnType(),
      docs: this.generatorMethodDocs(),
      statements: this.generatorMethodBody(),
    };
    return this.ast.generateMethodStatements(statement);
  }
}
