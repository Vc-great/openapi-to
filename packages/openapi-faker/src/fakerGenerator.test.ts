import { dirname } from "node:path";
import { fileURLToPath } from "node:url";

import { AST, OpenAPI } from "@openapi-to/core";

import _ from "lodash";
import Oas from "oas";
import path from "path";

import petStore from "../mock/petstore.json";
import { FakerGenerator } from "./FakerGenerator.ts";

import type { OpenapiToSingleConfig } from "@openapi-to/core";
const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

describe("faker", async () => {
  const openapiToSingleConfig: OpenapiToSingleConfig = {
    input: {
      path: "",
      name: "",
    },
    output: {
      dir: path.resolve(__dirname, "../mock"),
    },
    plugins: [],
  };

  const context = {
    output: "",
  };

  test("getFullText", () => {
    const ast = new AST();
    // @ts-expect-error Not a canonical document
    const oas = new Oas(petStore);
    const openapi = new OpenAPI({}, oas);
    const pluginConfig = {
      createZodDecorator: true,
      compare: false,
    };

    const requestGenerator = new FakerGenerator({
      oas,
      ast,
      openapi,
      pluginConfig,
      openapiToSingleConfig,
    });
    requestGenerator.build(context);
    const text = _.chain(ast.sourceFile)
      .map((sourceFile) => sourceFile.getFullText())
      .join("\n")
      .value();

    expect(text).toMatchSnapshot();
  });

  test("compare getFullText", () => {
    const ast = new AST();
    // @ts-expect-error Not a canonical document
    const oas = new Oas(petStore);
    const openapi = new OpenAPI({}, oas);
    const pluginConfig = {
      createZodDecorator: true,
      compare: true,
    };

    const requestGenerator = new FakerGenerator({
      oas,
      ast,
      openapi,
      pluginConfig,
      openapiToSingleConfig,
    });
    requestGenerator.build(context);
    const text = _.chain(ast.sourceFile)
      .map((sourceFile) => sourceFile.getFullText())
      .join("\n")
      .value();

    expect(text).toMatchSnapshot();
  });
});
