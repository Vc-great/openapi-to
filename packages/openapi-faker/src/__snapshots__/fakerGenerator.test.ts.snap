// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`faker > getFullText 1`] = `
"import { faker } from "@faker-js/faker";
import { testDto2, test32145, apiResponse, pet } from "./fakerModels";
import type { Pet } from "./Pet";

/**
 *
 * @tag pet
 * @description Everything about your Pets
 */
class PetFaker {
    /**
     *
     * @summary summary
     * @description
     */
    testPost(): NonNullable<Pet.TestPostResponse> {
        return testDto2()
    }

    /**
     *
     * @summary summary
     * @description
     */
    testPut(): NonNullable<Pet.TestPutResponse> {
        return testDto2()
    }

    /**
     *
     * @summary summary
     * @description
     */
    delByTest(): NonNullable<Pet.DelByTestResponse> {
        return test32145()
    }

    /**
     *
     * @summary summary
     * @description
     */
    testIdGet(): NonNullable<Pet.TestIdGetResponse> {
        return testDto2()
    }

    /**
     *
     * @summary uploads an image
     * @description pet
     */
    uploadImagePost(): NonNullable<Pet.UploadImagePostResponse> {
        return apiResponse()
    }

    /**
     *
     * @summary Add a new pet to the store
     * @description
     */
    create(): NonNullable<Pet.CreateResponse> {
        return {}
    }

    /**
     *
     * @summary Update an existing pet
     * @description
     */
    update(): NonNullable<Pet.UpdateResponse> {
        return {}
    }

    /**
     *
     * @summary Finds Pets by status
     * @description Multiple status values can be provided with comma separated strings
     */
    findByStatusGet(): NonNullable<Pet.FindByStatusGetResponse> {
        return faker.helpers.multiple(() => pet(), {
            count: 10,
        })
    }

    /**
     *
     * @summary Finds Pets by tags
     * @description Multiple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.
     */
    findByTagsGet(): NonNullable<Pet.FindByTagsGetResponse> {
        return faker.helpers.multiple(() => pet(), {
            count: 10,
        })
    }

    /**
     *
     * @summary Find pet by ID
     * @description Returns a single pet
     */
    findByPetId(): NonNullable<Pet.FindByPetIdResponse> {
        return pet()
    }

    /**
     *
     * @summary Updates a pet in the store with form data
     * @description
     */
    petIdPost(): NonNullable<Pet.PetIdPostResponse> {
        return {}
    }

    /**
     *
     * @summary Deletes a pet
     * @description
     */
    delByPetId(): NonNullable<Pet.DelByPetIdResponse> {
        return {}
    }
}

export const petFaker = new PetFaker;

import { faker } from "@faker-js/faker";
import { testDto2, test32145, apiResponse, pet, order } from "./fakerModels";
import type { Store } from "./Store";

/**
 *
 * @tag store
 * @description Access to Petstore orders
 */
class StoreFaker {
    /**
     *
     * @summary Place an order for a pet
     * @description
     */
    create(): NonNullable<Store.CreateResponse> {
        return order()
    }

    /**
     *
     * @summary Find purchase order by ID
     * @description For valid response try integer IDs with value >= 1 and <= 10. Other values will generated exceptions
     */
    findByOrderId(): NonNullable<Store.FindByOrderIdResponse> {
        return order()
    }

    /**
     *
     * @summary Delete purchase order by ID
     * @description For valid response try integer IDs with positive integer value. Negative or non-integer values will generate API errors
     */
    delByOrderId(): NonNullable<Store.DelByOrderIdResponse> {
        return {}
    }

    /**
     *
     * @summary Returns pet inventories by status
     * @description Returns a map of status codes to quantities
     */
    inventoryGet(): NonNullable<Store.InventoryGetResponse> {
        return {}
    }
}

export const storeFaker = new StoreFaker;

import { faker } from "@faker-js/faker";
import { testDto2, test32145, apiResponse, pet, order, user } from "./fakerModels";
import type { User } from "./User";

/**
 *
 * @tag user
 * @description Operations about user
 */
class UserFaker {
    /**
     *
     * @summary Creates list of users with given input array
     * @description
     */
    createWithArrayPost(): NonNullable<User.CreateWithArrayPostResponse> {
        return {}
    }

    /**
     *
     * @summary Creates list of users with given input array
     * @description
     */
    createWithListPost(): NonNullable<User.CreateWithListPostResponse> {
        return {}
    }

    /**
     *
     * @summary Get user by user name
     * @description
     */
    findByUsername(): NonNullable<User.FindByUsernameResponse> {
        return user()
    }

    /**
     *
     * @summary Updated user
     * @description This can only be done by the logged in user.
     */
    usernamePut(): NonNullable<User.UsernamePutResponse> {
        return {}
    }

    /**
     *
     * @summary Delete user
     * @description This can only be done by the logged in user.
     */
    delByUsername(): NonNullable<User.DelByUsernameResponse> {
        return {}
    }

    /**
     *
     * @summary Logs user into the system
     * @description
     */
    loginGet(): NonNullable<User.LoginGetResponse> {
        return
    }

    /**
     *
     * @summary Logs out current logged in user session
     * @description
     */
    logoutGet(): NonNullable<User.LogoutGetResponse> {
        return {}
    }

    /**
     *
     * @summary Create user
     * @description This can only be done by the logged in user.
     */
    create(): NonNullable<User.CreateResponse> {
        return {}
    }
}

export const userFaker = new UserFaker;

import { testdata } from "./testdata";
import { faker } from "@faker-js/faker";
import type { Test32145 } from "../models";

/** */
export function test32145(): Test32145 {
    return {
        /**description*/
        code: faker.number.int(),
        /***/
        meta: testdata()
    }
}

import { faker } from "@faker-js/faker";
import type { Test3214 } from "../models";

/** */
export function test3214(): Test3214 {
    return {
        /**默认值*/
        columnDefault: faker.string.alpha(),
        /**description*/
        columnLength: faker.number.int(),
        /**description*/
        columnName: faker.string.alpha(),
        /**description*/
        columnRemark: faker.string.alpha(),
        /**description*/
        columnScale: faker.number.int(),
        /**description*/
        columnType: faker.string.alpha(),
        /**description*/
        delFlag: faker.datatype.boolean(),
        /***/
        formId: faker.number.int(),
        /**id*/
        id: faker.number.int(),
        /**description*/
        notNull: faker.datatype.boolean()
    }
}

import { faker } from "@faker-js/faker";
import type { Testdata } from "../models";

/** */
export function testdata(): Testdata {
    return {
        /**description*/
        number: faker.number.int(),
        /**description*/
        numberOfElements: faker.number.int(),
        /**description*/
        totalElements: faker.number.int(),
        /**description*/
        totalPages: faker.number.int()
    }
}

import { faker } from "@faker-js/faker";
import type { Sort } from "../models";

/** */
export function sort(): Sort {
    return {
        /***/
        empty: faker.datatype.boolean(),
        /***/
        sorted: faker.datatype.boolean(),
        /***/
        unsorted: faker.datatype.boolean()
    }
}

import { test3214 } from "./test3214";
import { faker } from "@faker-js/faker";
import type { TestDto3 } from "../models";

/** */
export function testDto3(): TestDto3 {
    return {
        /***/
        test3214: faker.helpers.multiple(() => test3214(), {
            count: 10,
        })
    }
}

import { testDto3 } from "./testDto3";
import { testdata } from "./testdata";
import { faker } from "@faker-js/faker";
import type { TestDto2 } from "../models";

/** */
export function testDto2(): TestDto2 {
    return {
        /***/
        content: testDto3(),
        /***/
        meta: testdata()
    }
}

import { faker } from "@faker-js/faker";
import type { Test321 } from "../models";

/** */
export function test321(): Test321 {
    return {
        /***/
        id: faker.number.int()
    }
}

import { test321 } from "./test321";
import { faker } from "@faker-js/faker";
import type { TestDto } from "../models";

/** */
export function testDto(): TestDto {
    return {
        /**test321*/
        test321: faker.helpers.multiple(() => test321(), {
            count: 10,
        })
    }
}

import { apiResponse } from "./apiResponse";
import { faker } from "@faker-js/faker";
import type { RefList } from "../models";

/** */
export function refList(): RefList {
    return {
        /***/
        content: apiResponse(),
        /**string*/
        string: faker.string.alpha(),
        /***/
        name: faker.string.alpha(),
        /**title*/
        title: faker.string.alpha(),
        /**type*/
        type: faker.string.alpha()
    }
}

import { faker } from "@faker-js/faker";
import type { ApiResponse } from "../models";

/** */
export function apiResponse(): ApiResponse {
    return {
        /***/
        code: faker.number.int(),
        /***/
        type: faker.string.alpha(),
        /***/
        message: faker.string.alpha()
    }
}

import { faker } from "@faker-js/faker";
import type { Category } from "../models";

/** */
export function category(): Category {
    return {
        /***/
        id: faker.number.int(),
        /***/
        name: faker.string.alpha()
    }
}

import { category } from "./category";
import { tag } from "./tag";
import { faker } from "@faker-js/faker";
import type { Pet } from "../models";

/** */
export function pet(): Pet {
    return {
        /***/
        id: faker.number.int(),
        /***/
        category: category(),
        /***/
        name: faker.string.alpha(),
        /***/
        photoUrls: faker.helpers.multiple(() => faker.string.alpha(), {
            count: 10,
        }),
        /***/
        tags: faker.helpers.multiple(() => tag(), {
            count: 10,
        }),
        /**pet status in the store*/
        status: faker.string.alpha()
    }
}

import { faker } from "@faker-js/faker";
import type { Tag } from "../models";

/** */
export function tag(): Tag {
    return {
        /***/
        id: faker.number.int(),
        /***/
        name: faker.string.alpha()
    }
}

import { faker } from "@faker-js/faker";
import type { Order } from "../models";

/** */
export function order(): Order {
    return {
        /***/
        id: faker.number.int(),
        /***/
        petId: faker.number.int(),
        /***/
        quantity: faker.number.int(),
        /***/
        shipDate: faker.string.alpha(),
        /**Order Status*/
        status: faker.string.alpha(),
        /***/
        complete: faker.datatype.boolean()
    }
}

import { faker } from "@faker-js/faker";
import type { User } from "../models";

/** */
export function user(): User {
    return {
        /***/
        id: faker.number.int(),
        /***/
        username: faker.string.alpha(),
        /***/
        firstName: faker.string.alpha(),
        /***/
        lastName: faker.string.alpha(),
        /***/
        email: faker.string.alpha(),
        /***/
        password: faker.string.alpha(),
        /***/
        phone: faker.string.alpha(),
        /**User Status*/
        userStatus: faker.number.int()
    }
}

export * from "./testdata";
export * from "./test32145";
export * from "./test3214";
export * from "./sort";
export * from "./testDto3";
export * from "./testDto2";
export * from "./test321";
export * from "./testDto";
export * from "./apiResponse";
export * from "./refList";
export * from "./category";
export * from "./tag";
export * from "./pet";
export * from "./order";
export * from "./user";
"
`;
